test/validation/api/ipsec/ipsec.c

```diff
@@ -734,6 +746,44 @@ static int ipsec_send_out_one(const ipsec_test_part *part,
 	return num_out;
 }
 
+int ipsec_test_sa_update_seq_num(odp_ipsec_sa_t sa, uint32_t seq_num)
+{
+	odp_ipsec_test_sa_operation_t sa_op;
+	odp_ipsec_test_sa_param_t sa_param;
+
+	sa_op = ODP_IPSEC_TEST_SA_UPDATE_SEQ_NUM;
+	sa_param.seq_num = seq_num;
+
+	return odp_ipsec_test_sa_update(sa, sa_op, &sa_param);
+}
+
+static void ipsec_pkt_seq_num_check(odp_packet_t pkt, uint32_t seq_num)
+{
+	uint32_t l3_off = odp_packet_l3_offset(pkt);
+	uint32_t l4_off;
+	odph_ipv4hdr_t ip;
+
+	CU_ASSERT_NOT_EQUAL_FATAL(ODP_PACKET_OFFSET_INVALID, l3_off);
+	CU_ASSERT_EQUAL_FATAL(0, odp_packet_copy_to_mem(pkt, l3_off, sizeof(ip),
+							&ip));
+
+	if (ODPH_IPV4HDR_VER(ip.ver_ihl) == ODPH_IPV4) {
```

Comment:
Now this function silently succeeds without checking the sequence number if IP version is not 4 or if IP protocol is not ESP or AH (e.g. IPsec with UDP encapsulation). In those case the function should fail (CU_FAIL() or similar).

Reply:
Agreed. Fixed by invoking CU_FAIL() for unexpected IP version and IP proto values in v8.